src/include/miopen/pad_constant/problem_description.hpp

```diff
@@ -0,0 +1,173 @@
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(const TensorDescriptor& xDesc_,
+                       const TensorDescriptor& yDesc_,
+                       const size_t* padding_)
+        : xDesc(xDesc_), yDesc(yDesc_), padding(padding_)
+    {
+    }
+
+    const TensorDescriptor& GetXDesc() const { return xDesc; }
+    const TensorDescriptor& GetYDesc() const { return yDesc; }
+
+    NetworkConfig MakeNetworkConfig() const override;
+
+    bool IsSameType() const
+    {
+        if(xDesc.GetType() == yDesc.GetType())
+            return true;
+        return false;
+    }
+
+    bool IsSameShape() const
+    {
+        auto xSize = xDesc.GetSize();
+        auto ySize = yDesc.GetSize();
+        if(xSize == ySize)
+            return true;
+        return false;
+    }
+
+    bool checkContiguous(const TensorDescriptor& x) const
+    {
+        size_t s = 1;
+        for(int i = x.GetSize() - 1; i >= 0; --i)
+        {
+            if(s != x.GetStrides()[i])
+            {
+                return false;
+            }
+            s *= x.GetLengths()[i];
+        }
+        return true;
+    }
+
+    bool IsContiguous() const { return checkContiguous(xDesc) && checkContiguous(yDesc); }
+
+    bool IsImprovementOverROCm() const
+    {
+        if(IsContiguous())
+            // No contiguous case is faster
+            return false;
+        else
+            // Appears to be faster if we don't pad the first two
+            return padding[0] == 0 && padding[2] == 0;
+    }
+
+private:
+    const TensorDescriptor& xDesc;
+    const TensorDescriptor& yDesc;
+    const size_t* padding;
+};
+} // namespace pad_constant_fwd_contiguous
+
+namespace pad_constant_bwd {
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(const TensorDescriptor& xDesc_,
+                       const TensorDescriptor& yDesc_,
+                       const size_t* padding_)
+        : xDesc(xDesc_), yDesc(yDesc_), padding(padding_)
+    {
+    }
+
+    const TensorDescriptor& GetXDesc() const { return xDesc; }
+    const TensorDescriptor& GetYDesc() const { return yDesc; }
+
+    NetworkConfig MakeNetworkConfig() const override;
+
+    bool IsSameType() const
+    {
+        if(xDesc.GetType() == yDesc.GetType())
+            return true;
+        return false;
+    }
+
+    bool IsSameShape() const
+    {
+        auto xSize = xDesc.GetSize();
+        auto ySize = yDesc.GetSize();
+        if(xSize == ySize)
+            return true;
+        return false;
+    }
+
+    bool checkContiguous(const TensorDescriptor& x) const
+    {
+        size_t s = 1;
+        for(int i = x.GetSize() - 1; i >= 0; --i)
+        {
+            if(s != x.GetStrides()[i])
+            {
+                return false;
+            }
+            s *= x.GetLengths()[i];
+        }
+        return true;
+    }
+
+    bool IsContiguous() const { return checkContiguous(xDesc) && checkContiguous(yDesc); }
```

Comment:
You can use IsPacked function.

Reply:
Please check the conversation at https://github.com/moreh-dev/MIOpen/pull/12#discussion_r1597507827 on this

Reply:
We were missing about packed. Sorry. Please work by adding iscontiguous to tensor.cpp.

Reply:
I've added it as a method of TensorDescriptor